server.js

```diff
@@ -1031,6 +1024,41 @@ app.get("/rss", asyncMiddleware(async(req, res) => {
     res.sendFile(rss_path);
 }));
 
+app.get("/home", asyncMiddleware(async(req, res) => {
+    const client_list_24hr = await cacheIP24hr.wrap(
+        "IP24hr", "5m",
+        () => Promise.resolve(db.collection("games").distinct("ip", { _id: { $gt: objectIdFromDate(Date.now() - 1000 * 60 * 60 * 24) } })));
+
+    const client_list_1hr = await cacheIP1hr.wrap("IP1hr", "30s", () => Promise.resolve(db.collection("games").distinct("ip", { _id: { $gt: objectIdFromDate(Date.now() - 1000 * 60 * 60) } })));
+
+    const selfplay_24hr = await db.collection("games").find({ _id: { $gt: objectIdFromDate(Date.now() - 1000 * 60 * 60 * 24) } }).count();
+
+    const selfplay_1hr = await db.collection("games").find({ _id: { $gt: objectIdFromDate(Date.now() - 1000 * 60 * 60) } }).count();
+
+    const match_total = await db.collection("match_games").find().count();
+
+    const match_24hr = await db.collection("match_games").find({ _id: { $gt: objectIdFromDate(Date.now() - 1000 * 60 * 60 * 24) } }).count();
+
+    const match_1hr = await db.collection("match_games").find({ _id: { $gt: objectIdFromDate(Date.now() - 1000 * 60 * 60) } }).count();
+
+    const pug_data = {
+        matches: await dbutils.get_matches(db, 10),
```

Comment:
The current root page caches the match table data to avoid hitting the db on every load until new match is submitted. Not sure if that cache needs to be kept for this query.

Reply:
I do kept `cacheIP23hr` and` cacheIP1hr`. Any caches that I missed?

Reply:
The other cache is `cachematches`. Previously when we called database for match data, we wrapped it in the cachematches cache so we never ask database for match info again until the cache is cleared. Since get_matches can have various parameters, perhaps the cache wrapper should go in the await result right at this line?

Reply:
I see. I'll come up with a solution.